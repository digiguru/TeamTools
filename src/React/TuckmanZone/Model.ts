
export interface ITuckmanZone {
    focus      : string;
    visibility : "appearing" | "hiding";
    index      : number;
    label      : "forming" | "storming" | "norming" | "performing";
    maxWidth?  : number;
    username?  : string;
    Events?    : ITuckmanZoneEvents;
}
export interface ITuckmanZoneEvents {
  onZoneMouseDown: (zone: "forming" | "storming" | "norming" | "performing") => void;
  onZoneMouseUp  : (username: string, zone: "forming" | "storming" | "norming" | "performing", maxDistance: number, event: any) => void;
  onZoneOverFocus: (zone: "forming" | "storming" | "norming" | "performing") => void;
  onZoneOffFocus : (zone: "forming" | "storming" | "norming" | "performing") => void;
}

export interface ITuckmanZoneEventList {
    Events?: ITuckmanZoneEvents;
}
