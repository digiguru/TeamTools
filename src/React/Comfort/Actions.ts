import { IUserList, IUser } from "../User/Model";
import { InMemoryBrowserUsers } from "../../Shared/InMemoryBrowserUsers";


export const ComfortActions = {
    SET_STAGESIZE       : "SET_STAGESIZE",
    SET_STAGEVISIBILITY : "SET_STAGEVISIBILITY",
    SET_ZONEFOCUS       : "SET_ZONEFOCUS",
    SET_USERFOCUS       : "SET_USERFOCUS",
    SET_USERLIST        : "SET_USERLIST",
    SELECT_USER         : "SELECT_USER",
    CHOOSE_ZONE         : "CHOOSE_ZONE",
    TOGGLE_CHOICES      : "TOGGLE_CHOICES"
};

export function setStageSize(width: number, height: number) {
    return {type: ComfortActions.SET_STAGESIZE, width: width, height: height};
}
export function setStageVisibility(visibility: "hiding" | "appearing") {
    return {type: ComfortActions.SET_STAGEVISIBILITY, visibility: visibility};
}
export function setUserFocus(user: string, focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_USERFOCUS, user: user, focus: focus};
}
export function setZoneFocus(area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_ZONEFOCUS, area: area, focus: focus};
}
export function selectUser(user: string) {
    return {type: ComfortActions.SELECT_USER, user: user};
}


export function chooseZone(user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number) {
    return {type: ComfortActions.CHOOSE_ZONE, user: user, area: area, distance: distance};
}
export function toggleChoiceVisibility(visible: boolean) {
    return {type: ComfortActions.TOGGLE_CHOICES, visible: visible};
}

export function recieveUserList(userList: IUserList) {
    return {type: ComfortActions.SET_USERLIST, userList: userList};
}
export function fetchUserList() {
    return dispatch => {
        dispatch(recieveUserList({
            ShowUsers : false,
            Users : []
        }));
        const users = new InMemoryBrowserUsers(window);
        users.getUsers().then((data) => {
            const userList: IUser[] = data.map((v): IUser => {
                return { Username : v.name };
            });
            dispatch(recieveUserList({ShowUsers: true, Users: userList}));
        });
    };
    // return {type: ComfortActions.SET_USERLIST, userList: userList};
}