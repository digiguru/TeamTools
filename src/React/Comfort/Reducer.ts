
import {ComfortActions} from "./Actions";
import {ComfortAppState} from "./Model";
import { Point } from "../Models/Point";
import { Size } from "../Models/Size";
import { DOMMeasurement } from "../Models/IDomMeasurement";
import { IUser, IUserList } from "../User/Model";

const { fromJS, List, Map } = require('immutable');

const initialSize: Size = new Size(800, 800);
const initialState: ComfortAppState = {
    Size       : initialSize,
    CenterPoint: new Point(initialSize.width / 2, initialSize.height / 2),
    UserList   : {
    ShowUsers  : true,
    Users      : [
            {Username: "Adam Hall", Focus: "not-in-focus", Y: 0 },
            {Username: "Caroline Hall", Focus: "not-in-focus", Y: 0 }
        ]
    },
    Zones      : {
        Comfort: {Name: "Comfort", Focus: "not-in-focus", Range: {Start: 0, End: 33}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Stretch: {Name: "Stretch", Focus: "not-in-focus", Range: {Start: 34, End: 66}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Chaos  : {Name: "Chaos", Focus: "not-in-focus", Range: {Start: 67, End: 100}, Size: {Width: new DOMMeasurement("100%"), Height: new DOMMeasurement("100%")}}
    },
    ShowUserChoices: false,
    UserChoices    : []
};
export function comfortReducer(state: ComfortAppState = initialState, action): ComfortAppState {
    switch (action.type) {
        case ComfortActions.SET_USERLIST:
            return ComfortZoneAction.setUsers(state, (<any>action).userList);
        case ComfortActions.SET_STAGEVISIBILITY:
            return ComfortZoneAction.setVisibility(state, (<any> action).visibility);
        case ComfortActions.SET_STAGESIZE:
            return ComfortZoneAction.setStageSize(state, (<any>action).width, (<any>action).height);
        case ComfortActions.SET_USERFOCUS:
            return ComfortZoneAction.setUserFocus(state, (<any>action).user, (<any>action).focus);
        case ComfortActions.SET_ZONEFOCUS:
            return ComfortZoneAction.setZoneFocus(state, (<any>action).area, (<any>action).focus);
        case ComfortActions.SELECT_USER:
            return ComfortZoneAction.selectUser(state, (<any>action).user);
        case ComfortActions.CHOOSE_ZONE:
            return ComfortZoneAction.chooseZone(state, (<any>action).user, (<any>action).area, (<any>action).distance);
        case ComfortActions.TOGGLE_CHOICES:
            return ComfortZoneAction.toggleChoiceVisibility(state, (<any>action).visible);
        default:
            return state;
    }
}
class ComfortZoneAction {
    static setStageSize(state: ComfortAppState, width: number, height: number) {
        const newCenter = new Point(width / 2, height / 2);
        return fromJS(state)
            .set("Size", new Size(width, height))
            .set("CenterPoint", newCenter)
            .toJS();
    }
    static setVisibility(state: ComfortAppState, visibility: "hiding" | "appearing") {
        return fromJS(state)
            .setIn(["Zones", "Comfort", "visibility"], visibility)
            .setIn(["Zones", "Stretch", "visibility"], visibility)
            .setIn(["Zones", "Chaos", "visibility"], visibility).toJS();
        //  state.Zones.Chaos.visibility
    }

    static setZoneFocus(state: ComfortAppState, area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        return fromJS(state)
            .setIn(["Zones", "Comfort", "Focus"], area === "Comfort" ? focus : "not-in-focus")
            .setIn(["Zones", "Stretch", "Focus"], area === "Stretch"  ? focus : "not-in-focus")
            .setIn(["Zones", "Chaos", "Focus"], area === "Chaos"  ? focus : "not-in-focus").toJS();
    }

    static setUserFocus(state: ComfortAppState, user: string, focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        const originalList = List(state.UserList.Users);
        const newUserList = originalList.update(
            originalList.findIndex(item => item.Username === user),
            (item) => fromJS(item).set("Focus", focus)).toJS();
        return fromJS(state)
            .setIn(["UserList", "Users"], newUserList).toJS();
    }
    static selectUser(state: ComfortAppState, user: String): ComfortAppState {
        const originalList = List(state.UserList.Users);
        const item: IUser = originalList.find(item => item.Username === user);
        // Sets currentUser, and therefor hides the user choice menu
        const data = fromJS(state)
            .set("CurrentUser", item)
            .set("ShowUserChoices", false)
            .setIn(["UserList", "ShowUsers"], false);

        return data.toJS();
    }

    static setUsers(state: ComfortAppState, userList: IUserList): ComfortAppState {
        console.log("Input", state.UserList);
        console.log("Update", userList);

        const data = fromJS(state)
            .set("UserList", userList).toJS();
        console.log("Output", data.UserList);
        return data;
    }

    static chooseZone(
        state: ComfortAppState, user: string, area: "Chaos" | "Stretch" | "Comfort",
        distance: number): ComfortAppState {

        // Add the user choice
        const newUserChoices = List(state.UserChoices).push({
            User: {Username: user},
            Zone: area,
            Distance: distance
        }).toJS();
        // Remove the user from the choice list
        const newUserList = List(state.UserList.Users).filter((item) => item.Username !== user).toArray();
        // Show the user list
        const showUserChoice = !!(newUserList.length);
        // Return
        return fromJS(state)
            .delete("CurrentUser")
            .set("ShowUserChoices", showUserChoice)
            .set("UserChoices", newUserChoices)
            .setIn(["UserList", "Users"], newUserList)
            .setIn(["UserList", "ShowUsers"], showUserChoice).toJS();
    }
    static toggleChoiceVisibility(state: ComfortAppState, visible: boolean): ComfortAppState {
        // Set "showUserChoices" to true
        return Map(state)
            .set("ShowUserChoices", visible).toJS();
    }
}
