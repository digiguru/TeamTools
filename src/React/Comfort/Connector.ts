import { connect } from "react-redux";
import { ComfortStage } from "./Component";
import { ComfortUserChoiceState } from "./Model";
import { setStageVisibility } from "./Actions";

const mapStateToProps = (state: ComfortUserChoiceState): ComfortUserChoiceState => {
    return state;
};

const mapDispatchToProps = (dispatch): any => {
    return {
        Events: {
            onHide: (): void => {
                dispatch(setStageVisibility("hiding"));
            },
            onShow: (): void => {
                dispatch(setStageVisibility("appearing"));
            }
        }
    };
};
export const ComfortConnector = connect(
    mapStateToProps,
    mapDispatchToProps
)(ComfortStage);
