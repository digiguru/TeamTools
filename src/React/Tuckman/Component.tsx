import * as React from "react";
import { ITuckmanModel } from "./Model";
import { ReduxUserConnector } from "../User/Connector";
import { ITuckmanZone } from "../TuckmanZone/Model";
import { TuckmanZoneConnector } from "../TuckmanZone/Connector";
import { TuckmanUserHistoryConnector } from "../TuckmanUserChoice/Connector";


export const TuckmanStage = (state: ITuckmanModel) => {
    return <g>
                <g id="zones">
                    <TuckmanZoneConnector label="performing"  />
                    <TuckmanZoneConnector label="norming"     />
                    <TuckmanZoneConnector label="storming"    />
                    <TuckmanZoneConnector label="forming"     />
                </g>
                <ReduxUserConnector />
                <TuckmanUserHistoryConnector />
        <button id="show-tuckman" onMouseUp={state.onShow}>show tuckman</button>
        <button id="hide-tuckman" onMouseUp={state.onHide}>hide tuckman</button>
    </g>;
};
