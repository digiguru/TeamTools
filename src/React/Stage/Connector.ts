import { connect } from "react-redux";
import { Stage } from "./Component";
import { IStageState, IStageProps } from "./Model";

export const StageConnector = connect(
  (state: IStageState, props: IStageProps) => {
    return { Size: state.Size, InnerBits: props.children};
  }
)(Stage);
