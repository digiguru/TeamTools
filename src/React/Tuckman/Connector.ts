import { connect } from "react-redux";
import { TuckmanStage } from "./Component";
import { ITuckmanModel } from "./Model";
import { IHidableEvents } from "../Stage/Model";
import { setStageVisibility } from "./Actions";

const mapStateToProps = (state: ITuckmanModel): ITuckmanModel => {
    return state;
};

const mapDispatchToProps = (dispatch): IHidableEvents => {
    return {
        Events: {
            onHide: (): void => {
                dispatch(setStageVisibility("hiding"));
            },
            onShow: (): void => {
                dispatch(setStageVisibility("appearing"));
            }
        }
    };
};
export const TuckmanConnector = connect(
    mapStateToProps,
    mapDispatchToProps
)(TuckmanStage);
