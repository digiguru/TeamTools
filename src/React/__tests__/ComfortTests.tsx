import * as React from "react";
import { createStore } from "redux";
import { ComfortStage } from "../Comfort/Component";
import {comfortReducer} from "../Comfort/Reducer";
// import { render } from "react-dom";
import { Provider } from "react-redux";
import * as Action from "../Comfort/Actions";
import { StageConnector } from "../Stage/Connector";
import { IUserList } from "../User/Model";


const renderizer = require("react-test-renderer");

test("Should not mutate in any way", () => {
    const myState = comfortReducer(undefined, {type: "Startup"});
    const initialState = JSON.stringify(myState);
    const checkAfterAction = (action) => {
        const currentState = comfortReducer(myState, action);
        expect(initialState).toEqual(JSON.stringify(myState));
    };
    checkAfterAction(Action.setUserFocus("Adam Hall", "in-focus"));
    checkAfterAction(Action.selectUser("Adam Hall"));
    checkAfterAction(Action.setStageSize(800, 600));
    checkAfterAction(Action.setZoneFocus("Chaos", "in-focus"));
    checkAfterAction(Action.setZoneFocus("Stretch", "active"));
    checkAfterAction(Action.setZoneFocus("Comfort", "in-focus"));
    checkAfterAction(Action.toggleChoiceVisibility(true));
    checkAfterAction(Action.toggleChoiceVisibility(false));
    checkAfterAction(Action.chooseZone("Adam Hall", "Stretch", 85));
});

function renderStore(store) {
    return renderizer.create(
        <svg xmlns="http://www.w3.org/2000/svg" id="stage">
            <Provider store={store}>
                <StageConnector>
                    <ComfortStage />
                </StageConnector>
            </Provider>
        </svg>
    );
}

test("Should show the component", () => {
    // Arrange
    const myStore = createStore(comfortReducer);

  

    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setUserFocus("Adam Hall", "in-focus"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

});

test("Should allow shrinking", () => {
    // Arrange

    const myStore = createStore(comfortReducer);

   

    myStore.dispatch(Action.chooseZone("Adam Hall", "Stretch", 50));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

    myStore.dispatch(Action.chooseZone("Caroline Hall", "Chaos", 100));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

});


test("Should allow hiding", () => {
    // Arrange

    const myStore = createStore(comfortReducer);

    

    myStore.dispatch(Action.setStageVisibility("hiding"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

    myStore.dispatch(Action.setStageVisibility("appearing"));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();

});



test("Should allow users to be set okay", () => {
    // Arrange

    const myStore = createStore(comfortReducer);

   
    const users: IUserList = {
        Users: [
            {Username: "Test person 1"},
            {Username: "Test person 2"},
            {Username: "Test person 3"}
        ]
    };
    myStore.dispatch(Action.recieveUserList(users));
    expect(renderStore(myStore).toJSON()).toMatchSnapshot();
});



