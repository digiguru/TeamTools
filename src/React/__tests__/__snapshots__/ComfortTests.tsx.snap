// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should allow hiding 1`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="appear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Adam Hall
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={150}
            />
            <text
              className="username"
              x="60"
              y={210}
            >
              Caroline Hall
            </text>
          </g>
        </g>
        <g
          id="history"
        />
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should allow hiding 2`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="appear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Adam Hall
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={150}
            />
            <text
              className="username"
              x="60"
              y={210}
            >
              Caroline Hall
            </text>
          </g>
        </g>
        <g
          id="history"
        />
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should allow shrinking 1`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="appear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Caroline Hall
            </text>
          </g>
        </g>
        <g
          id="history"
        >
          <circle
            className="point"
            cx={800}
            cy={400}
            r="10"
          />
        </g>
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should allow shrinking 2`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="disappear"
          id="users"
        />
        <g
          id="history"
        >
          <circle
            className="point"
            cx={800}
            cy={400}
            r="10"
          />
          <circle
            className="point"
            cx={-400}
            cy={399.99999999183456}
            r="10"
          />
        </g>
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should allow users to be set okay 1`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="disappear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Test person 1
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={150}
            />
            <text
              className="username"
              x="60"
              y={210}
            >
              Test person 2
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={240}
            />
            <text
              className="username"
              x="60"
              y={300}
            >
              Test person 3
            </text>
          </g>
        </g>
        <g
          id="history"
        />
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should show the component 1`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="appear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Adam Hall
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={150}
            />
            <text
              className="username"
              x="60"
              y={210}
            >
              Caroline Hall
            </text>
          </g>
        </g>
        <g
          id="history"
        />
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;

exports[`Should show the component 2`] = `
<svg
  id="stage"
  xmlns="http://www.w3.org/2000/svg"
>
  <svg
    height={800}
    id="stage"
    width={800}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <g>
        <g>
          <rect
            className="not-in-focus"
            height="100%"
            id="chaos"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            width="100%"
          />
          <text
            className="area-label"
            id="label-chaos"
            textAnchor="middle"
            x="50%"
            y="20"
          >
            chaos
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="33%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="20%"
          >
            stretch
          </text>
        </g>
        <g>
          <circle
            className="not-in-focus"
            cx="50%"
            cy="50%"
            height="50%"
            id="stretch"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            r="15%"
            width="50%"
          />
          <text
            className="area-label"
            id="label-stretch"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            comfort
          </text>
        </g>
        <g
          className="appear"
          id="users"
        >
          <g
            className="user-group"
          >
            <rect
              className="in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={60}
            />
            <text
              className="username"
              x="60"
              y={120}
            >
              Adam Hall
            </text>
          </g>
          <g
            className="user-group"
          >
            <rect
              className="not-in-focus"
              height="90"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              width="800"
              x="0"
              y={150}
            />
            <text
              className="username"
              x="60"
              y={210}
            >
              Caroline Hall
            </text>
          </g>
        </g>
        <g
          id="history"
        />
      </g>
      ,
      <button
        id="show-tuckman"
      >
        show tuckman
      </button>
      ,
      <button
        id="hide-tuckman"
      >
        hide tuckman
      </button>
    </g>
  </svg>
</svg>
`;
